### DISCUSSION TITLE
Ticket  not recognized.

### DISCUSSION THREAD

**Discussion by a-rnam** [akunzai/GSS.Authentication.CAS#530] [Q&A] [answered]
Hi experts! :)

Could anyone help me with following error:

info: GSS.Authentication.CAS.AspNetCore.CasAuthenticationHandler`1[[GSS.Authentication.CAS.AspNetCore.CasAuthenticationOptions, GSS.Authentication.CAS.AspNetCore, Version=5.3.3.0, Culture=neutral, PublicKeyToken=51dae670fa3140f8]][4]
      Error from RemoteAuthentication: Ticket 'ST-81561-P1ebbTmrg2FBOKKNYU3lt1NObyo-cas-2' not recognized.
fail: Microsoft.AspNetCore.Server.IIS.Core.IISHttpServer[2]
      Connection ID "15780613108801732819", Request ID "800000d4-0003-db00-b63f-84710c7967bb": An unhandled exception was thrown by the application.
      System.Exception: An error was encountered while handling the remote login.
       ---> System.Security.Authentication.AuthenticationException: Ticket 'ST-81561-P1ebbTmrg2FBOKK

The error appears when the site is loaded for a first time.
![image](https://github.com/akunzai/GSS.Authentication.CAS/assets/127849782/cd839533-33bb-4f0f-8586-1bd89855b852)

 After refreshing the site everything works fine. What could the reason of this?

Kind regards,
Peter

**Comment by akunzai** [comment, marked as answer]
The error message was response from your CAS server when service-ticket validation failed

It was throws at [HERE](https://github.com/akunzai/GSS.Authentication.CAS/blob/2e829be54f6e1dd05979d9f2c01744dd8809396b/src/GSS.Authentication.CAS.Core/Validation/Cas20ServiceTicketValidator.cs#L47)

Perhaps you should check `web.config` inheritance problems like [THIS](https://github.com/akunzai/GSS.Authentication.CAS/discussions/426) if this problem only happens on IIS

- https://stackoverflow.com/q/782252
- https://learn.microsoft.com/aspnet/core/host-and-deploy/iis/web-config#configuration-of-aspnet-core-module-with-webconfig
